DEVELOPERS.md

```diff
@@ -33,6 +33,45 @@ Dependencies are managed with `uv`.
 
 ### Overview
 See the [uv documentation](https://docs.astral.sh/uv/concepts/projects/dependencies) for details on usage.
+Commands for adding, removing or modifying constraints of dependencies will automatically respect the
+global timestamp cutoff specified in the `pyproject.toml`:
+```toml
+[tool.uv]
+exclude-newer = "YYYY-MM-DDTHH:MM:SSZ"
+```
+Note that `uv.lock` must be reproducible from `pyproject.toml`. Otherwise, `just check` will fail.
+If `just check` errors saying that the timestamps must match, you might have modified one file but not the other:
+  - If you modified `pyproject.toml`, you must update `uv.lock` via `uv lock` / `just upgrade-all` or similar.
+  - If you did not modify `pyproject.toml` but have changes in `uv.lock`, you should revert the changes to `uv.lock`,
```

Comment:
I think maybe this should have an extra warning that `uv.lock` should never be manually edited